### PR TITLE
Fix explicit method type with inherited declaration

### PR DESCRIPTION
**Pull request by fo5for**
Small update to correct the semantics before we work on refactoring the mess 😅

### CODE REVIEW

File: shared/src/test/diff/mlscript/Methods2.mls [hunk 1]
```diff
@@ -68,22 +67,22 @@ l.Size
 l.Map (fun x -> mul x 2)
 //│ res: List['B .. 'B | int]
 
-o = l.HeadOption
```
Comment (LPTK) [inline shared/src/test/diff/mlscript/Methods2.mls:71]: Small detail: let's not swap these needlessly.

**Review by LPTK — APPROVED** [review]
LGTM. Please go ahead and do the squash-merge.

File: shared/src/test/diff/mlscript/Methods2.mls (hunk 1 above)
Comment (fo5for) [reply shared/src/test/diff/mlscript/Methods2.mls:71]: The pattern matching below only fails with an `Option`. If `Cons.HeadOption` is called explicitly, the result would be a `Some` and the match would succeed. So I swapped them to preserve the error below.

Comment (LPTK) [reply shared/src/test/diff/mlscript/Methods2.mls:71]: Oh yes good point, I had missed that 👍

**Pull request merged**